exchanges/binance/binance_types.go

```diff
@@ -17,12 +18,1018 @@ const (
+	PositionSide    string  `json:"positionSide"`
+	Buyer           bool    `json:"buyer"`
+	Maker           bool    `json:"maker"`
+}
+
+// FuturesIncomeHistoryData stores futures income history data
+type FuturesIncomeHistoryData struct {
+	Symbol     string  `json:"symbol"`
+	IncomeType string  `json:"incomeType"`
+	Income     float64 `json:"income,string"`
+	Asset      string  `json:"asset"`
+	Info       string  `json:"info"`
+	Timestamp  int64   `json:"time"`
+}
+
+// NotionalBracketData stores notional bracket data
+type NotionalBracketData struct {
+	Pair     string `json:"pair"`
+	Brackets []struct {
+		Bracket          int64   `json:"bracket"`
+		InitialLeverage  float64 `json:"initialLeverage"`
+		QtyCap           float64 `json:"qtyCap"`
+		QtylFloor        float64 `json:"qtyFloor"`
+		MaintMarginRatio float64 `json:"maintMarginRatio"`
+	}
+}
+
+// ForcedOrdersData stores forced orders data
+type ForcedOrdersData struct {
+	OrderID       int64   `json:"orderId"`
+	Symbol        string  `json:"symbol"`
+	Status        string  `json:"status"`
+	ClientOrderID string  `json:"clientOrderId"`
+	Price         float64 `json:"price,string"`
+	AvgPrice      float64 `json:"avgPrice,string"`
+	OrigQty       float64 `json:"origQty,string"`
+	ExecutedQty   float64 `json:"executedQty,string"`
+	CumQuote      float64 `json:"cumQuote,string"`
+	TimeInForce   string  `json:"timeInForce"`
+	OrderType     string  `json:"orderType"`
+	ReduceOnly    bool    `json:"reduceOnly"`
+	ClosePosition bool    `json:"closePosition"`
+	Side          string  `json:"side"`
+	PositionSide  string  `json:"positionSide"`
+	StopPrice     float64 `json:"stopPrice,string"`
+	WorkingType   string  `json:"workingType"`
+	PriceProtect  float64 `json:"priceProtect,string"`
+	OrigType      string  `json:"origType"`
+	Time          int64   `json:"time"`
+	UpdateTime    int64   `json:"updateTime"`
+}
+
+// ADLEstimateData stores data for ADL estimates
+type ADLEstimateData struct {
+	Symbol      string `json:"symbol"`
+	ADLQuantile struct {
+		Long  float64 `json:"LONG"`
+		Short float64 `json:"SHORT"`
+		Hedge float64 `json:"HEDGE"`
+	} `json:"adlQuantile"`
+}
+
+// InterestHistoryData gets interest history data
+type InterestHistoryData struct {
+	Asset       string  `json:"asset"`
+	Interest    float64 `json:"interest"`
+	LendingType string  `json:"lendingType"`
+	ProductName string  `json:"productName"`
+	Time        string  `json:"time"`
+}
+
+// FundingRateData stores funding rates data
+type FundingRateData struct {
+	Symbol      string  `json:"symbol"`
+	FundingRate float64 `json:"fundingRate,string"`
+	FundingTime int64   `json:"fundingTime"`
+}
+
+// SymbolsData stores perp futures' symbols
+type SymbolsData struct {
+	Symbol string `json:"symbol"`
+}
+
+// PerpsExchangeInfo stores data for perps
+type PerpsExchangeInfo struct {
+	Symbols []SymbolsData `json:"symbols"`
+}
+
+// UFuturesExchangeInfo stores exchange info for ufutures
+type UFuturesExchangeInfo struct {
```

Comment:
Safe to ignore this comment... but...
Here's another potential split of source files for types. Like `ufutures_types.go` etc...
Actually, I wouldn't put the types separate from the api functions in the first place - never seen it done before - e.g. I'd put this type in the `ufutures.go`, but for some reason this is the convention used in this project

Normally the convention for splitting to separate files - a single .go file contains related functionality for a single concept. Types and functions for these types are very related, so jumping back and forth in the source files to understand what is being done doesn't really help for readability.